### ISSUE TITLE
Run tasks from root directory

### ISSUE THREAD

**Issue by jaredpalmer** [vercel/turborepo#845]
### Describe the feature you'd like to request

As a user, I want to be able to cache tasks defined in the root `package.json` and cache them.

```jsonc
{
   "pipeline": {
      "_root#build": {
          "inputs": ["list/of/inputs/to/hash/**"],
          "outputs": ["some/output/dir/**"],
          "dependsOn":["prepare"]
      },
      "_root#prepare": {
          "inputs": ["foo/**"],
          "outputs": ["another/output/dir/**"],
      },
      ....
}      
```

### Describe the solution you'd like

-  https://github.com/vercel/turborepo/issues/523
	- Consider migrating env vars to `inputs` and out of `dependsOn`? idk? 
- Update task graph construction to include root task edges in scheduler https://github.com/vercel/turborepo/blob/main/cli/internal/core/scheduler.go#L183
  - We may also need to adjust resolving root deps, perhaps we can simply add the root to `PackageInfos` and clean up specialized lockfile resolution https://github.com/vercel/turborepo/blob/main/cli/internal/context/context.go#L164

### Describe alternatives you've considered

None

**Comment by jlarmstrongiv** [comment]
That would be very nice!  I have dozens of manual scripts to do that for me

`"turbo:npm-check-updates:validate:all": "npx turbo run turbo:npm-check-updates:validate && npm run turbo:npm-check-updates:validate",`

**Comment by silawrenc** [comment]
👍 We would also like this because I think it would give us an easy way to run tasks in parallel at the root. For example we have a CI task which basically runs `eslint` and `jest` both from the root of the monorepo. Ideally we'd run these in parallel. At the moment we'd use something like [concurrently](https://www.npmjs.com/package/concurrently) for this, but it's always nice to rationalise our tooling 😄.

**Comment by dagda1** [comment]
@gsoltis I noticed [this PR](https://github.com/vercel/turborepo/pull/1040#pullrequestreview-939899052) this got merged into the main repo.  This might be the answer to my prayers :).

When you say

> That is, turbo.json must have an entry for _root#<task

What is `_root` in this context?  I have tried using the root package.json name but that does not work.

**Comment by gsoltis** [comment]
@dagda1 Apologies, I still need to put up the corresponding docs PR for this feature. `_root` was the initial token we used to denote tasks run from the root package, but after implementing, elected to switch to `//`. So, you need `//#<task>` in `turbo.json`

**Comment by dagda1** [comment]
@gsoltis no need to apologise, than you for your work!

**Comment by dagda1** [comment]
@gsoltis is it `//task` or `//#task`, i.e. you need the `#`

**Comment by gsoltis** [comment]
`//#task`. `//` is the token meaning "root package", `#` is the separator between "package" and "task". So just like building the `ui` package would be `ui#build`, running `build` in the root package is `//#build`

**Comment by dagda1** [comment]
@gsoltis at last, I have full turbo.  Thank you sir
<img width="310" alt="tsc" src="https://user-images.githubusercontent.com/118328/169101669-8f9a4e01-d502-4f3f-981b-f5cc0932ceb0.png">

**Comment by gsoltis** [comment]
Awesome! Sounds like you don't need it at this point, but the docs PR is #1260

**Comment by dagda1** [comment]
@gsoltis why would I not need it?

**Comment by gsoltis** [comment]
Oh, I just meant that it seemed like you had gotten it working
